epub33/rs/index.html

```diff
@@ -1604,23 +1604,46 @@ <h4>Initial containing block dimensions</h4>
 									data-tests="#lay-fxl-xhtml-icb,#lay-fxl-xhtml-icb_multi">They MUST clip content
 									positioned outside of the ICB.</span>
 							</p>
+
 							<p id="confreq-fxl-rs-xhtml-icb-units" data-tests="#lay-fxl-xhtml-icb_units">If the width or
 								height values in the <code>viewport meta</code> tag contain a non-numeric character but
 								start with a number (e.g., the value includes a unit of length declaration such as
 								"500px"), the number prefix SHOULD be used as the pixel value, otherwise the value MUST
 								be treated as invalid.</p>
-							<p>If the <code>viewport meta</code> does not contain a width or a height value, or if these
-								values are invalid, reading systems SHOULD consider these as having the values
-									<code>device-width</code> and <code>device-height</code>, respectively.</p>
+
+							<p>
+								If the <code>viewport meta</code> tag is invalid, the reading system MAY attempt to
+								extract valid <code>width</code> and <code>height</code> values and, if successful, use those to establish
+								the initial containing block.
+							</p>
+							
+							<p>
+								If the <code>viewport meta</code> does not contain a width or a height value, or if these
+								values are invalid, reading systems MAY choose to:
+							</p>
+
+							<ul>
+								<li>consider these as having the values <code>device-width</code> and <code>device-height</code>, respectively; or</li>
+								<li>look for the previous content document in the [=spine=], if applicable, and use the ICB value defined there; or</li>
+								<li>consider the content document as erroneous XHTML content altogether.</li>
+							</ul>
+
 							<p id="confreq-fxl-rs-xhtml-multiple-icb-def"
 								data-tests="#lay-fxl-xhtml-icb_multi_declarations">If an XHTML content document contains
-								more than one <code>viewport meta</code> tag, reading systems MUST use the first in
+								more than one <code>viewport meta</code> tags, reading systems MUST use the first in
 								document order to obtain the height and width dimensions. Subsequent declarations MUST
 								be ignored.</p>
+
 							<p id="confreq-fxl-rs-xhtml-icb-ratio">When the ICB aspect ratio does not match the aspect
 								ratio of the reading system [=content display area=], reading systems MAY position the
 								ICB inside the area to accommodate the user interface; in other words, added
 								letter-boxing space MAY appear on either side (or both) of the content.</p>
+
+							<p>
+								In all cases when the <code>viewport meta</code> tag is invalid, the reading system MAY issue 
```

Comment:
Isn't this true for all errors? Feels like it could be stated somewhere more general for error handling.

Reply:
Yes, this is my last remark in the issue description. There are a number of other places in the spec that refer to errors explicitly, although all are referring to package errors. There seems to be no mention of a generic error handling, though, for example, what happens if the XHTML is invalid?

Maybe, at this point, the best would be to raise a separate issue/PR for error handling statement and then remove this from the current PR.

Reply:
Something more general on error handling is likely a good idea. I know in practice, when RS run into XHTML or parsing errors, we just throw the usual browser error page.